### PR TITLE
Stack function

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented stack function.
It is equivalent to stack method in numpy.
http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.stack.html

### CODE REVIEW

**Comment by delta2323** [comment]
Please check the behavior when NumPy is older than 1.10.0

**Comment by unnonouno** [comment]
I found that `concat` doesn't support negative axes.

**Comment by unnonouno** [comment]
I found that split function has bug. #1282

**Comment by unnonouno** [comment]
Negative axes are now supported. Please review the PR.

File: chainer/functions/array/stack.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from chainer.functions.array import concat
+from chainer.functions.array import expand_dims
+
+
+def stack(xs, axis=0):
+    """Concatenate variables along a new axis.
+
+    Args:
+        xs (list of chainer.Variable): Variables to be concatenated.
+        axis (int): Axis of result along which variables stacked.
```
Comment (delta2323) [inline chainer/functions/array/stack.py]: variables *are* stacked

Comment (delta2323) [inline chainer/functions/array/stack.py]: `of result` is not needed.

File: tests/chainer_tests/functions_tests/array_tests/test_stack.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    @attr.gpu
+    def test_forward_gpu(self):
+        self.check_forward([cuda.to_gpu(x) for x in self.xs])
+
+    def check_backward(self, xs_data, g_data):
+        def func(*xs):
+            return functions.stack(xs, self.axis)
+
+        gradient_check.check_backward(
+            func, xs_data, g_data, eps=2.0 ** -2, atol=1e-3, rtol=1e-3)
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/array_tests/test_stack.py]: Is the value of `eps` equivalent to `4e-2`?

**Comment by delta2323** [comment]
Could you consider applying this [comment](https://github.com/pfnet/chainer/pull/1262#issuecomment-226373939)? I would like to check not only the shape but also the value of the resultant even if NumPy is older than 1.10.0.

**Comment by unnonouno** [comment]
I checked its values: https://github.com/pfnet/chainer/pull/1262/files#diff-440df1a13fc241a19aa1a40cd569074fR48

**Comment by delta2323** [comment]
> I checked its value: https://github.com/pfnet/chainer/pull/1262/files#diff-440df1a13fc241a19aa1a40cd569074fR48

Sorry, I missed it.

File: tests/chainer_tests/functions_tests/array_tests/test_stack.py (hunk 2 above)
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/array_tests/test_stack.py]: Yes. We can easily modify its scale with this notation.

Comment (delta2323) [inline tests/chainer_tests/functions_tests/array_tests/test_stack.py]: OK, make sense.

**Comment by delta2323** [comment]
LGTM 👍 . I will merge it after Travis passed.

**Pull request merged**